crates/bevy_ui/src/render/ui_material_pipeline.rs

```diff
@@ -436,140 +449,144 @@ pub fn prepare_uimaterial_nodes<M: UiMaterial>(
     mut ui_meta: ResMut<UiMaterialMeta<M>>,
     mut extracted_uinodes: ResMut<ExtractedUiMaterialNodes<M>>,
     view_uniforms: Res<ViewUniforms>,
+    globals_buffer: Res<GlobalsBuffer>,
     ui_material_pipeline: Res<UiMaterialPipeline<M>>,
     mut phases: Query<&mut RenderPhase<TransparentUi>>,
     mut previous_len: Local<usize>,
 ) {
     if let Some(view_binding) = view_uniforms.uniforms.binding() {
-        let mut batches: Vec<(Entity, UiMaterialBatch<M>)> = Vec::with_capacity(*previous_len);
-
-        ui_meta.vertices.clear();
-        ui_meta.view_bind_group = Some(render_device.create_bind_group(
-            "ui_material_view_bind_group",
-            &ui_material_pipeline.view_layout,
-            &BindGroupEntries::single(view_binding),
-        ));
-        let mut index = 0;
-
-        for mut ui_phase in &mut phases {
-            let mut batch_item_index = 0;
-            let mut batch_shader_handle = AssetId::invalid();
-
-            for item_index in 0..ui_phase.items.len() {
-                let item = &mut ui_phase.items[item_index];
-                if let Some(extracted_uinode) = extracted_uinodes.uinodes.get(item.entity) {
-                    let mut existing_batch = batches
-                        .last_mut()
-                        .filter(|_| batch_shader_handle == extracted_uinode.material);
-
-                    if existing_batch.is_none() {
-                        batch_item_index = item_index;
-                        batch_shader_handle = extracted_uinode.material;
-
-                        let new_batch = UiMaterialBatch {
-                            range: index..index,
-                            material: extracted_uinode.material,
-                        };
-
-                        batches.push((item.entity, new_batch));
+        if let Some(globals_binding) = globals_buffer.buffer.binding() {
```

Comment:
a minor suggestion - if you use `if let (Some(view_binding), Some(globals_binding)) = (view_uniforms.uniforms.binding(), globals_buffer.buffer.binding()) {` we can cut the diff in half :)